frontend/src/components/Inquiries/InquiriesTable.tsx

```diff
@@ -29,38 +31,76 @@ const InquiriesTable = ({ themes }: InquiriesTableProps) => {
   const { data: inquiries } = useInquiries()
 
   const [sortedInquiries, setSortedInquiries] = useState<InquiryPublic[]>([])
+
   useEffect(() => {
-    setSortedInquiries([])
-    if (inquiries?.data) {
-      inquiries?.data.sort((a: InquiryPublic, b: InquiryPublic) => {
-        return (
-          (schedule?.scheduled_inquiries?.indexOf(a.id) ?? -1) -
-          (schedule?.scheduled_inquiries?.indexOf(b.id) ?? -1)
+    if (inquiries) {
+      const inquiriesList = inquiries.data
+        .filter(
+          (i) => (schedule?.scheduled_inquiries?.indexOf(i.id) ?? -1) >= -1,
         )
-      })
-      setSortedInquiries([...inquiries.data])
+        .sort((a: InquiryPublic, b: InquiryPublic) => {
+          return (
+            (schedule?.scheduled_inquiries?.indexOf(a.id) ?? -1) -
+            (schedule?.scheduled_inquiries?.indexOf(b.id) ?? -1)
+          )
+        })
+      setSortedInquiries([...inquiriesList])
     }
-  }, [schedule, inquiries?.data])
+  }, [schedule, inquiries])
```

Comment:
Shouldn't this logic be in `useMemo` instead of `useEffect`? We only want the calculations to run when schedule or inquiries change. `useEffect` is meant for situations where you need to do something after rendering, such as API calls, logging, or interacting with the DOM.